python/tskit/trees.py

```diff
@@ -366,11 +370,33 @@ def __init__(
         self.id = id_
         self.site = site
         self.node = node
+        self.time = time
         self.derived_state = derived_state
         self.parent = parent
         self._encoded_metadata = encoded_metadata
         self._metadata_decoder = metadata_decoder
 
+    def __eq__(self, other):
+        # We need to remove metadata and the decoder so we are just comparing
+        # the encoded metadata, along with the other attributes.
+        # We also need to remove time as have to compare to unknown time.
+        other_ = {**other.__dict__}
```

Comment:
Maybe ``other_ = copy.copy(other)`` is a bit more readable?

Reply:
Needs to be `other_ = copy.copy(other).__dict__` but is clearer, changed in https://github.com/tskit-dev/tskit/pull/672/commits/bc4041b63c354897d8f4c0d7538bd80592880beb